src/utils/regex.cc

```diff
@@ -311,5 +378,43 @@ int Regex::search(const std::string& s) const {
 #endif
 }
 
+unsigned long Regex::get_default_match_limit() const {
+    unsigned long default_match_limit;
+#ifdef WITH_PCRE2
+    int ret = pcre2_config(PCRE2_CONFIG_MATCHLIMIT, &default_match_limit);
```

Comment:
The doc says this should be a pointer to a uint32_t.

(Of course, this only needs to be addressed if we don't abandon this get_default functionality entirely).